libr/bin/obj.c

```diff
@@ -8,21 +8,148 @@
 	if (binfile->rbin->verbose) \
 	eprintf
 
+static void mem_free(void *data) {
+	RBinMem *mem = (RBinMem *)data;
+	if (mem && mem->mirrors) {
+		mem->mirrors->free = mem_free;
+		r_list_free (mem->mirrors);
+		mem->mirrors = NULL;
+	}
+	free (mem);
+}
+
+static void object_delete_items(RBinObject *o) {
+	ut32 i = 0;
+	r_return_if_fail (o);
+	sdb_free (o->addr2klassmethod);
+	r_list_free (o->entries);
+	r_list_free (o->fields);
+	r_list_free (o->imports);
+	r_list_free (o->libs);
+	r_list_free (o->relocs);
+	r_list_free (o->sections);
+	r_list_free (o->strings);
+	r_list_free (o->symbols);
+	r_list_free (o->classes);
+	r_list_free (o->lines);
+	sdb_free (o->kv);
+	if (o->mem) {
+		o->mem->free = mem_free;
+	}
+	r_list_free (o->mem);
+	o->mem = NULL;
+	o->entries = NULL;
+	o->fields = NULL;
+	o->imports = NULL;
+	o->libs = NULL;
+	o->relocs = NULL;
+	o->sections = NULL;
+	o->strings = NULL;
+	o->symbols = NULL;
+	o->classes = NULL;
+	o->lines = NULL;
+	o->info = NULL;
+	o->kv = NULL;
+	for (i = 0; i < R_BIN_SYM_LAST; i++) {
+		free (o->binsym[i]);
+		o->binsym[i] = NULL;
+	}
+}
+
 R_IPI void r_bin_object_free(void /*RBinObject*/ *o_) {
 	RBinObject *o = o_;
 	if (!o) {
 		return;
 	}
 	free (o->regstate);
 	r_bin_info_free (o->info);
-	r_bin_object_delete_items (o);
+	object_delete_items (o);
 	R_FREE (o);
 }
 
+static char *swiftField(const char *dn, const char *cn) {
+	char *p = strstr (dn, ".getter_");
+	if (!p) {
+		p = strstr (dn, ".setter_");
+		if (!p) {
+			p = strstr (dn, ".method_");
+		}
+	}
+	if (p) {
+		char *q = strstr (dn, cn);
+		if (q && q[strlen (cn)] == '.') {
+			q = strdup (q + strlen (cn) + 1);
+			char *r = strchr (q, '.');
+			if (r) {
+				*r = 0;
+			}
+			return q;
+		}
+	}
+	return NULL;
+}
+
+static RList *classes_from_symbols(RBinFile *bf, RBinObject *o) {
+	RBinSymbol *sym;
+	RListIter *iter;
+	RList *symbols = o->symbols;
+	RList *classes = o->classes;
```

Comment:
what about not taking this and always create a new list?, the one who calls this function should be in charge to fill o->classes?

For consistency maybe it should be like this,otherwise the function name may change. If we decide to just updae the binobj inside each of those calls instead of returning a list. im fine too. but it can be useful at some point to have access to those apis, not just having them internally.